pkg/syncer/spec/spec_process.go

```diff
@@ -181,17 +202,33 @@ func (c *Controller) ensureDownstreamNamespaceExists(ctx context.Context, downst
 		})
 	}
 
-	// TODO(sttts): check that namespace exists in lister before using the client
-	if _, err := namespaces.Create(ctx, newNamespace, metav1.CreateOptions{}); err != nil {
-		// An already exists error is ok - it means something else beat us to creating the namespace.
-		if !k8serrors.IsAlreadyExists(err) {
-			// Any other error is not good, though.
-			// TODO bubble this up as a condition somewhere.
-			klog.Errorf("Error while creating namespace %q: %v", downstreamNamespace, err)
+	// Check if the namespace already exists, if not create it.
+	namespace, err := c.downstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}).Lister().Get(newNamespace.GetName())
+	if err != nil && apierrors.IsNotFound(err) {
+		if _, err := namespaces.Create(ctx, newNamespace, metav1.CreateOptions{}); err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				klog.Infof("Using already existing downstream namespace %s for upstream namespace %s|%s", newNamespace.GetName(), desiredNSLocator.Workspace, desiredNSLocator.Namespace)
+				return nil
```

Comment:
You have to fall through to the "The namespace exists, so check if it has the correct namespace locator." code.

Reply:
I reverted this change. If I get an error, I throw an error to enqueue the item again; if something created the namespace before us, we would pick it in the next run by reading the namespaceLocator. Seems easier than adding another Get here.